packages/astro/src/env/vite-plugin-env.ts

```diff
@@ -47,11 +48,8 @@ export function astroEnv({
 				fileURLToPath(settings.config.root),
 				'',
 			);
-			for (const [key, value] of Object.entries(loadedEnv)) {
-				if (value !== undefined) {
-					process.env[key] = value;
-				}
-			}
+			(globalThis as any)[ENV_SYMBOL] ??= {};
+			Object.assign((globalThis as any)[ENV_SYMBOL], loadedEnv);
```

Comment:
Should we directly assign it like
```suggestion
			(globalThis as any)[ENV_SYMBOL] ??=  loadedEnv;
```

Since no one else should be messing with the symbol